service/group0_state_broadcaster.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "group0_state_broadcaster.hh"
+#include "compaction/compaction_manager.hh"
+#include "replica/database.hh"
+
+namespace service {
+
+namespace {
+
+logging::logger slogger("group0_state_broadcaster");
+
+} // namespace
+
+void group0_state_broadcaster::start() {
```

Comment:
Judging from the code below, this is not only broadcasting the state (it should be "state_id" instead of "state"), but also collecting the states and updating repair time.

Reply:
moved, names now as "group0_state_id_handler"